doc/source/schema/schema.md

```diff
@@ -36,33 +39,73 @@ The terminology used in this specification is based on [xarray's terminology](ht
 `InferenceData` stores all quantities that are relevant to fulfilling its goals in different groups. Different groups generally distinguish conceptually different quantities in Bayesian inference, however, convenience in {ref}`creation <creating_InferenceData>` and {ref}`usage <working_with_InferenceData>` of `InferenceData` objects also plays a role. In general, each quantity (such as posterior distribution or observed data) will be represented by several multidimensional labeled variables.
 
 ### Rules
-Following are a few rules which should be followed:
+Below are a few rules which should be followed:
 * Each group should have one entry per variable and each variable should be named.
-* When relevant, the first two dimensions of each variable should be the sample identifier (`chain`, `draw`).
-* For groups like `observed_data` or `constant_data`, the two initial dimensions(`chain`, `draw`) are omitted.
+* Dimension names `chain`, `draw`, `sample` and `post_pred_id` are reserved for
+  InferenceData use to indicate sample dimensions.
+* Dimensions in InferenceData (including sample dimensions) should be identified by name only.
+* For groups like `observed_data` or `constant_data`, all sample dimensions can be
+  omitted. For groups like `prior`, `posterior` or `posterior_predictive` either `sample` has to be
+  present or both `chain` and `draw` dimensions need to be present. Any combinations that follow
+  this are valid.
 * Dimensions must be named and share name with a coordinate specifying the index values, called coordinate values.
 * Coordinate values can be repeated and should not necessarily be numerical values.
 * Variables must not share names with dimensions.
-* Moreover, each group contains the following attributes:
-  - `created_at`: the date of creation of the group.
-  - `inference_library`: the library used to run the inference.
-  - `inference_library_version`: version of the inference library used.
+* Groups, variables or the InferenceData itself can have arbitrary metadata stored.
 
-### Relations
-`InferenceData` data objects contain any combination of the groups described below. There are also some relations (detailed below) between the variables and dimensions of different groups. Hence, whenever related groups are present they should comply with these relations.
+### Metadata
+No metadata is _required_ to be present in order to be compliant with the InferenceData schema.
+However, it is recommended to store the following fields when relevant:
+* `name`: InferenceData objects represent multiple quantities related to Bayesian modelling,
+  but they are all tied to a single model. The model identifier can be added as metadata
+  to simplify the calls to model comparison functions.
+* `created_at`: the date of creation of the group.
+* `arviz_version`: the version of the ArviZ library that generated the InferenceData
+* `arviz_language`: the programming language from which ArviZ was used to create the InferenceData
```

Comment:
Could these be a bit more general?

If another program creates the InferenceData objects, it would be a good idea to save its version, language. So also the library name for it might be a good idea.

`creation_library`

Reply:
I 2nd this, along with a `creation_library_version`. We might also want `creation_library_language`. On the Julia side we could append a `".jl"` to the name if a user provides a package object itself instead of just the name to differentiate, but `".py"` for Python makes less sense.

Reply:
Yes, that is indeed much better, I'll update it